srv/components/misc_pkgs/kafka/config.sls

```diff
@@ -106,3 +106,34 @@ Update connections in kafka config:
     - pattern: ^zookeeper.connect=.*
     - repl: zookeeper.connect={{ node_hosts|join(',')}}
     - append_if_not_found: True
+
+{% if 1 < server_nodes|length %}
```

Comment:
Just to reconfirm, this condition is true when number of nodes are 3 (or more than 1) ?

Reply:
Yes
```
if 1 less than length of list (server_nodes)
```
So it is not specific to 3 nodes, but any number more than 1.

@sachinpunadikar 
I was thinking should the same logic be used to define the values for 
>default.replication.factor=3
>offsets.topic.replication.factor=3
>transaction.state.log.replication.factor=3

Reply:
@ypise thanks for explanation.
The config values are 3 till the time we are keeping kafka cluster limited to 3 nodes. Variation in kafka cluster size will have different config values and those will be derived by running some tests.
Hence current logic is good. 
kafka_setup will be useful to handle such cases in future.